### ISSUE TITLE
merge-layers Incompatible geometry types even with force

### ISSUE THREAD

**Issue by 2803media** [mbloch/mapshaper#380]
Hi I have plenty of json, here an example of the problem with 2 of them:

file1.json
```
{"type":"FeatureCollection", "features": [
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2951264265775873,49.148785981338975]},"properties":{"DEPCOM":"14319","TYPEQU":"A301","QUALITE_XY":"Mauvaise","LIB_MOD":"Réparation automobile et de matériel agricole"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.31152074425466153,49.15404831477448]},"properties":{"DEPCOM":"14319","TYPEQU":"A301","QUALITE_XY":"Bonne","LIB_MOD":"Réparation automobile et de matériel agricole"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2894776550113281,49.146241008077155]},"properties":{"DEPCOM":"14319","TYPEQU":"A301","QUALITE_XY":"Acceptable","LIB_MOD":"Réparation automobile et de matériel agricole"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.31152074425466153,49.15404831477448]},"properties":{"DEPCOM":"14319","TYPEQU":"A301","QUALITE_XY":"Bonne","LIB_MOD":"Réparation automobile et de matériel agricole"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.3105915934027392,49.15466041224897]},"properties":{"DEPCOM":"14319","TYPEQU":"A301","QUALITE_XY":"Mauvaise","LIB_MOD":"Réparation automobile et de matériel agricole"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14319","TYPEQU":"A401","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Maçon"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14319","TYPEQU":"A401","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Maçon"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14319","TYPEQU":"A403","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Menuisier charpentier serrurier"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14319","TYPEQU":"A403","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Menuisier charpentier serrurier"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14319","TYPEQU":"A404","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Plombier couvreur chauffagiste"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14319","TYPEQU":"A404","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Plombier couvreur chauffagiste"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.30830503106893314,49.15484559726204]},"properties":{"DEPCOM":"14319","TYPEQU":"A405","QUALITE_XY":"Bonne","LIB_MOD":"Électricien"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.30348012306719974,49.15193143855802]},"properties":{"DEPCOM":"14319","TYPEQU":"A405","QUALITE_XY":"Bonne","LIB_MOD":"Électricien"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2893635049621799,49.1444444177928]},"properties":{"DEPCOM":"14319","TYPEQU":"B309","QUALITE_XY":"Mauvaise","LIB_MOD":"Droguerie quincaillerie bricolage"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2854837145732062,49.14814985907869]},"properties":{"DEPCOM":"14319","TYPEQU":"D232","QUALITE_XY":"Mauvaise","LIB_MOD":"Infirmier"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2854837145732062,49.14814985907869]},"properties":{"DEPCOM":"14319","TYPEQU":"D232","QUALITE_XY":"Mauvaise","LIB_MOD":"Infirmier"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2892730729714769,49.146220596321285]},"properties":{"DEPCOM":"14319","TYPEQU":"E101","QUALITE_XY":"Bonne","LIB_MOD":"Taxi-VTC"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2914392090203291,49.150291139132044]},"properties":{"DEPCOM":"14319","TYPEQU":"E101","QUALITE_XY":"Bonne","LIB_MOD":"Taxi-VTC"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2928649998615454,49.15024689826918]},"properties":{"DEPCOM":"14319","TYPEQU":"F103","QUALITE_XY":"Bonne","LIB_MOD":"Tennis"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.29258000026454856,49.15103999811405]},"properties":{"DEPCOM":"14319","TYPEQU":"F113","QUALITE_XY":"Bonne","LIB_MOD":"Terrains de grands jeux"}},
{"type":"Feature","geometry":{"type":"Point","coordinates":[-0.2938520895245828,49.15163637866983]},"properties":{"DEPCOM":"14319","TYPEQU":"F117","QUALITE_XY":"Bonne","LIB_MOD":"Roller-Skate-Vélo-Bicross ou freestyle"}}
]}
```

file2.json:

```
{"type":"FeatureCollection", "features": [
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14062","TYPEQU":"A403","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Menuisier charpentier serrurier"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14062","TYPEQU":"A404","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Plombier couvreur chauffagiste"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14062","TYPEQU":"A404","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Plombier couvreur chauffagiste"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14062","TYPEQU":"A404","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Plombier couvreur chauffagiste"}},
{"type":"Feature","geometry":null,"properties":{"DEPCOM":"14062","TYPEQU":"A404","QUALITE_XY":"type_équipement_non_géolocalisé_cette_année","LIB_MOD":"Plombier couvreur chauffagiste"}}
]}
```

You see than in the second one the geometries are null 

I tried to merge those 2 files and got an error:

`/usr/bin/mapshaper -i file1.json file2json combine-files -merge-layers force -o result.json`

`[merge-layers] Error: Incompatible geometry types: [none], point`

The logic will be that the merge only keep the first file? What do you think?

**Comment by mbloch** [comment]
Hi, thanks for reporting this. The incompatible geometry error is actually intended to prevent combining polygons, lines and points in the same layer. (Mapshaper only allows one geometry type in a layer).  Mapshaper should be able to combine null shapes and a single geometry type, however, so the error you reported is a bug. I'll fix it now :)

**Comment by 2803media** [comment]
You're welcome thanks for improving this great tool!

**Comment by mbloch** [comment]
The fix for this issue is contained in the just-released 0.4.142 version, so I'm closing this.
